miden-crypto/src/dsa/rpo_falcon512/hash_to_point.rs

```diff
@@ -63,9 +63,21 @@ pub fn hash_to_point_shake256(message: &[u8], nonce: &Nonce) -> Polynomial<Falco
         reader.read(&mut randomness);
         let t = ((randomness[0] as u32) << 8) | (randomness[1] as u32);
         if t < K * MODULUS as u32 {
-            coefficients.push(FalconFelt::new((t % MODULUS as u32) as i16));
+            coefficients.push(u32_to_falcon_felt(t));
         }
     }
 
     Polynomial { coefficients }
 }
+
+// HELPER FUNCTIONS
+// ================================================================================================
+
+fn felt_to_falcon_felt(value: Felt) -> FalconFelt {
+    FalconFelt::new((value.as_int() % MODULUS as u64) as i16)
+}
+
+#[cfg(test)]
+fn u32_to_falcon_felt(value: u32) -> FalconFelt {
+    FalconFelt::new((value % MODULUS as u32) as i16)
```

Comment:
In `FalconFelt::new()` we do a bunch of manipulations with the passed-in value - could we add some doc comments (for the `FalconFlet` constructor) to explain what's going on there?

Reply:
I have added a comment explaining that this is mainly driven by the fact that the constructor accepts `i16`.